src/nullablearray.jl

```diff
@@ -3,40 +3,6 @@ import Base: convert, getindex, setindex!, similar, in
 ## Constructors and converters
 ## (special methods for AbstractArray{Union{T, Null}}, to avoid wrapping nulls inside CategoricalValues)
 
-CategoricalArray{Union{T, Null}, N, R}(dims::NTuple{N,Int}; ordered=false) where {T, N, R} =
-    CategoricalArray{Union{T, Null}, N, R}(zeros(R, dims), CategoricalPool{T, R}(ordered))
-
-CategoricalArray{Union{CategoricalValue{T, R}, Null}, N, R}(dims::NTuple{N,Int};
-                                                                      ordered=false) where {T, N, R} =
-    CategoricalArray{Union{T, Null}, N, R}(dims; ordered=ordered)
-CategoricalArray{Union{CategoricalValue{T}, Null}, N, R}(dims::NTuple{N,Int};
-                                                                   ordered=false) where {T, N, R} =
-    CategoricalArray{Union{T, Null}, N, R}(dims; ordered=ordered)
-CategoricalArray{Union{CategoricalValue{T, R}, Null}, N}(dims::NTuple{N,Int};
-                                                                   ordered=false) where {T, N, R} =
-    CategoricalArray{Union{T, Null}, N, R}(dims; ordered=ordered)
-CategoricalArray{Union{CategoricalValue{T}, Null}, N}(dims::NTuple{N,Int};
-                                                                ordered=false) where {T, N} =
-    CategoricalArray{Union{T, Null}, N}(dims; ordered=ordered)
-# CategoricalArray{Union{CategoricalValue, Null}, N}}){N}(dims::NTuple{N,Int};
-#                                                                 ordered=false) =
-#     CategoricalArray{Union{String, Null}, N}(dims; ordered=ordered)
-# CategoricalArray{Union{CategoricalValue, Null}}}){N}(dims::NTuple{N,Int};
-#                                                              ordered=false) =
-#     CategoricalArray{Union{String, Null}, N}(dims; ordered=ordered)
-
-CategoricalVector{T}(m::Int; ordered=false) where {T>:Null} =
-    CategoricalArray{T}((m,); ordered=ordered)
-CategoricalMatrix{T}(m::Int, n::Int; ordered=false) where {T>:Null} =
-    CategoricalArray{T}((m, n); ordered=ordered)
-
-CategoricalArray(A::AbstractArray{T}; ordered=_isordered(A)) where {T>:Null} =
-    CategoricalArray{T}(A, ordered=ordered)
-CategoricalVector(A::AbstractVector{T}; ordered=_isordered(A)) where {T>:Null} =
-    CategoricalVector{T}(A, ordered=ordered)
-CategoricalMatrix(A::AbstractMatrix{T}; ordered=_isordered(A)) where {T>:Null} =
-    CategoricalMatrix{T}(A, ordered=ordered)
-
 function convert(::Type{CategoricalArray{T, N, R}}, A::AbstractArray{S, N}) where {S, T>:Null, N, R}
```

Comment:
I wonder whether this constructor is still needed too now that you take care of unwrapping in the low-level constructor.